src/sass/parse.js

```diff
@@ -1115,15 +1115,30 @@ function getBrackets() {
 function checkClass(i) {
   let start = i;
   let l;
+  let wasIdent = true;
 
   if (i >= tokensLength) return 0;
 
   if (tokens[i].class_l) return tokens[i].class_l;
 
   if (tokens[i++].type !== TokenType.FullStop) return 0;
 
-  if (l = checkIdentOrInterpolation(i)) i += l;
-  else return 0;
+  // If a class is prefixed with `-` only allow if it contains an ident later
+  if (tokens[i].type === TokenType.HyphenMinus) wasIdent = false;
+
+  while (i < tokensLength) {
+    if (l = checkIdentOrInterpolation(i)) {
+      wasIdent = true;
+      i += l;
+    }
+    else if (tokens[i].type === TokenType.HyphenMinus) i += 1;
+    else break;
+  }
```

Comment:
According to CSS validator, this is a valid class name: `-a---a--`, but this is not: `--a`. It means that there can be any number of hyphens just not at the beginning. To make the code above disallow situation with multiple hyphens, I'd propose removing `wasIdent` variable and making something like that:
```js
// Check for `-` at beginning.
if (tokens[i].type === TokenType.HyphenMinus) i+= 1;

// Original lines.
if (l = checkIdentOrInterpolation(i)) i += l;
else return 0;

// Here goes your loop, just without `wasIdent` thing.
while (...) {...}
```